src/ORM/Behavior/Translate/TranslateStrategyTrait.php

```diff
@@ -0,0 +1,190 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Behavior\Translate;
+
+use Cake\Datasource\EntityInterface;
+use Cake\Event\EventInterface;
+use Cake\I18n\I18n;
+use Cake\ORM\Table;
+
+/**
+ * Contains common code needed by TranslateBehavior strategy classes.
+ */
+trait TranslateStrategyTrait
+{
+
+    /**
+     * Table instance
+     *
+     * @var \Cake\ORM\Table
+     */
+    protected $table;
+
+    /**
+     * The locale name that will be used to override fields in the bound table
+     * from the translations table
+     *
+     * @var string|null
+     */
+    protected $locale;
+
+    /**
+     * Instance of Table responsible for translating
+     *
+     * @var \Cake\ORM\Table
+     */
+    protected $translationTable;
+
+    /**
+     * Return translation table instance.
+     *
+     * @return \Cake\ORM\Table
+     */
+    public function getTranslationTable(): Table
```

Comment:
Would it be enough to typehint against the RepositoryInterface? Do we need any methods from concrete Table class?

Reply:
These strategies are not going to work for non relational databases. Methods for setting associations, getting 
primary key etc. don't exist in `RepositoryInterface`.

Reply:
Maybe we should introduce `TableInterface` then?

Reply:
That would be cleaner :)

Reply:
Why not, but that's beyond the scope of this PR.